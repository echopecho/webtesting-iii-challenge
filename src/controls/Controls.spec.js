import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import renderer from 'react-test-renderer';

import Controls from './Controls';

describe('the Controls component', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<Controls />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match the snapshot when closed', () => {
    const tree = renderer.create(<Controls closed={true} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match the snapshot when not closed', () => {
    const tree = renderer.create(<Controls close={false} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match the snapshot when locked', () => {
    const tree = renderer.create(<Controls locked={true} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should match the snapshot when not locked', () => {
    const tree = renderer.create(<Controls locked={false} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});