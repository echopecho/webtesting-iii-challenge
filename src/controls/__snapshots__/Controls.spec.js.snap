// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Controls component should match the snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
  >
    Close Gate
  </button>
</div>
`;

exports[`the Controls component should match the snapshot when closed 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={false}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
  >
    Open Gate
  </button>
</div>
`;

exports[`the Controls component should match the snapshot when locked 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Unlock Gate
  </button>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Close Gate
  </button>
</div>
`;

exports[`the Controls component should match the snapshot when not closed 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
  >
    Close Gate
  </button>
</div>
`;

exports[`the Controls component should match the snapshot when not locked 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    disabled={false}
  >
    Close Gate
  </button>
</div>
`;
